src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -43,4 +51,25 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
 
         logger.info("Challenge with id: " + challengeRepository.save(challenge).getId() + " has been saved");
     }
+
+    @Override
+    public Set<ChallengeDTO> getChallengesByUserId(long id) throws EmptyChallengeSetException {
+
+        Set<ChallengeDTO> userChallengesSet = new LinkedHashSet<>();
+
+        for(UserChallenge userChallenge : userChallengeRepository.findAllByUserId(id))
```

Comment:
userChallengeRepository.findAllByUserId(id) ar trebui sa declari separat o variabila care sa contina set-ul returnat de aceasta metoda. Nu face apelulu in bucla de for, nu este un good practice. In felul acesta este greu de urmarit de cinva care-ti citeste codul.